### PR TITLE
Add controller method and tests to Repository Branches

### LINKED ISSUE #7: Integration tests for branch resource
- [ ] Create Tests on spec file

- [ ] Create controller methods for the test

### PR DESCRIPTION
**Pull request by MariethPerez**
Close #7

### CODE REVIEW

File: spec/controllers/branches_controller_spec.rb [hunk 1]
```diff
@@ -1,4 +1,298 @@
 require 'rails_helper'
 
 describe BranchesController do
+    #   before do
+    #     Product.delete_all
+    #   end
```
Comment (diegotc86) [inline spec/controllers/branches_controller_spec.rb]: Hi Marieth, this lines wasn't needed?

File: spec/controllers/branches_controller_spec.rb [hunk 2]
```diff
@@ -1,4 +1,298 @@
...
+          )
+          repository = Repository.create(
+            name: 'repo1', 
+            description: 'Description of repo1', 
+            access: "public", 
+            license: "none", 
+            user_id: user.id
+          )
+          post :create, params: { name: "Rama_Feature1",user_id: user.id , repository_id: repository.id }
+          puts response.body
```
Comment (diegotc86) [inline spec/controllers/branches_controller_spec.rb]: Maybe this "puts" is not neccesary

File: spec/controllers/branches_controller_spec.rb (hunk 1 above)
Comment (MariethPerez) [reply spec/controllers/branches_controller_spec.rb]: Yes, this lines weren't necessary. I will take them out.

File: spec/controllers/branches_controller_spec.rb (hunk 2 above)
Comment (MariethPerez) [reply spec/controllers/branches_controller_spec.rb]: Thanks,  I will take them out.

**Pull request merged**